nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2032,10 +2038,14 @@ protected Object getDataModelField(DocumentRef docRef, String schema, String fie
     @Override
     public void makeRecord(DocumentRef docRef) {
         Document doc = resolveReference(docRef);
-        if (isRecord(docRef)) {
+        checkPermission(doc, READ);
+        if (doc.isRecord()) {
```

Comment:
This sounds like a refactoring, shouldn't it be in a separate commit?

Reply:
Done